### ISSUE TITLE
BorderStyle support in Box Component

### ISSUE THREAD

**Issue by Gurmanjot** [razorpay/blade#1438]
`borderStyle` prop is not supported by the Box component. Currently the only way to get a border is to pass boxWidth- Top/down/left/right and not with boxWidth itself. 

Adding either of the props `borderStyle` or `boxWidth` to hasBorder constant ([here](https://github.com/razorpay/blade/pull/1182/files#r1200085423)) will work.

**Comment by saurabhdaware** [comment]
> boxWidth
You mean `borderWidth`?

Added it in this PR https://github.com/razorpay/blade/pull/1361#discussion_r1259182853. Will be resolved when it is merged.

Can also create a separate PR if you're blocked on this.

**Comment by Gurmanjot** [comment]
> 

Yes, borderWidth.

**Comment by chaitanyadeorukhkar** [comment]
Raising a separate PR for this since its a P0 #1445 .
